src/main/java/io/debezium/connector/yugabytedb/connection/OpId.java

```diff
@@ -126,12 +126,18 @@ public int hashCode() {
     @Override
     public int compareTo(OpId o) {
         // Unsigned comparison
-        if (term != o.term)
-            return term + Long.MIN_VALUE < o.term + Long.MIN_VALUE ? -1 : 1;
-        else if (index != o.index)
-            return index + Long.MIN_VALUE < o.index + Long.MIN_VALUE ? -1 : 1;
-        else
-            return write_id + Long.MIN_VALUE < o.write_id + Long.MIN_VALUE ? -1 : 1;
+        if (term != o.term) {
```

Comment:
This can be reverted since we are using the `isLesserThanOrEqualTo` check to ensure that the logic to compare the checkpoints remain the same everywhere.